internal/controllers/topology/cluster/scope/hookresponsetracker.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package scope
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+
+	runtimehooksv1 "sigs.k8s.io/cluster-api/exp/runtime/hooks/api/v1alpha1"
+)
+
+// HookResponseTracker is a helper to capture the responses of the various lifecycle hooks.
+type HookResponseTracker struct {
+	responses map[string]runtime.Object
+}
+
+// NewHookResponseTracker returns a new HookResponseTracker.
+func NewHookResponseTracker() *HookResponseTracker {
+	return &HookResponseTracker{
+		responses: map[string]runtime.Object{},
+	}
+}
+
+// Add add the response of a hook to the tracker.
+func (h *HookResponseTracker) Add(hook string, response runtime.Object) {
+	h.responses[hook] = response
+}
+
+// EffectiveRequeueAfter calculates the lowest non-zero retryAfterSeconds time from all the tracked responses.
```

Comment:
WDYT about naming those funcs:
* EffectiveRetryAfter
* EffectiveMessage
then they are consistent with each other and clearly map to the fields in the response